### PR TITLE
Add strftime variant with background color

### PR DESCRIPTION
**Pull request by mikosoft83**
Add strftime variant with background color (useful for antialiased fonts with non black background)

# What does this implement/fix?

Adds strftime variant with background color (useful for antialiased fonts with non black background) 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Other

**Related issue or feature (if applicable):**

- fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):**

- esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by jesserockz** [comment]
Please write a meaningful PR title

File: esphome/components/display/display.h [hunk 1]
```diff
@@ -447,6 +447,19 @@ class Display : public PollingComponent {
...
+  void strftime(int x, int y, BaseFont *font, Color color, Color background, TextAlign align, const char *format,
+                ESPTime time) __attribute__((format(strftime, 8, 0)));
+
+  /** Evaluate the strftime-format `format` and print the result with the top left at [x,y] with `font`.
+   *
+   * @param x The x coordinate of the upper left corner.
+   * @param y The y coordinate of the upper left corner.
+   * @param font The font to draw the text with.
+   * @param color The color to draw the text with.
+   * @param background The background color to draw the text with.
```
Comment (jesserockz) [inline esphome/components/display/display.h]: This is on the wrong function

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by mikosoft83 — COMMENTED** [review]
Added the align param

**Comment by mikosoft83** [comment]
Is this still missing something?

File: esphome/components/display/display.h (hunk 1 above)
Comment (jesserockz) [reply esphome/components/display/display.h]: This is still on the wrong function. You put the new params as the doc for the function that doesnt have them

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/display/display.h (hunk 1 above)
Comment (mikosoft83) [reply esphome/components/display/display.h]: I had a brain rot. Hopefully it's all fixed now.

**Comment by mikosoft83** [comment]
I'm just bumping this

**Comment by jesserockz** [comment]
> I'm just bumping this

Unfortunately, you never fixed the comment on the function correctly, I was waiting for you to actually fix it. I have now fixed it myself

**Comment by mikosoft83** [comment]
Ah, I now see what the issue was. I apologize, I very rarely post any PRs as I'm not familiar with GitHub, but I wanted to help since posting issues usually takes forever to resolve.
Thank you for your patience.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**